### PR TITLE
src/dmclock: bring in fixes for indirect_intrusive_heap

### PR DESCRIPTION
**Pull request by athanatos**

### CODE REVIEW

**Comment by athanatos** [comment]
http://pulpito.ceph.com/sjust-2019-12-20_10:21:53-rados-wip-sjust-dmclock-fix-master-distro-basic-smithi

Above run filtered for centos 8 -- other failures appear unrelated/similar to master runs.

**Comment by athanatos** [comment]
https://github.com/ceph/dmclock/pull/75 needs to be merged first.

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -108,6 +108,12 @@ if(CMAKE_CXX_COMPILER_ID STREQUAL GNU)
   endif()
 endif(CMAKE_CXX_COMPILER_ID STREQUAL GNU)
 
+CHECK_C_COMPILER_FLAG("-D_GLIBCXX_ASSERTIONS" HAS_GLIBC_DEBUG)
```
Comment (jdurgin) [inline src/CMakeLists.txt]: may want to restrict this to debug builds

**Comment by athanatos** [comment]
@jdurgin Turns out there are 3 unittests that don't pass with GLIBCXX_ASSERTIONS enabled.  I've dropped that commit for now and will fix those in a separate PR.

**Comment by jdurgin** [comment]
sounds good

**Pull request merged**